### PR TITLE
Jg/test semgrep

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**
A clear and concise description of the features you're adding in this pull request.

**Additional context**
Add any other context about the problem you're solving.

**Metadata**
- Fixes #[Link to Issue]

### CODE REVIEW

File: .semgrepignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+# Semgrep rules folder
+.semgrep
+
+# Semgrep-action log folder
+.semgrep_logs/
+
+l2geth/
+packages/*/node_modules
+packages/*/test
```
[the remark this answers is not in the document]
Comment (trianglesphere) [reply .semgrepignore:23]: I didn't realize we had those. I don't like those defaults though. If we want a newline at EOF, it should be a lint.

[the remark this answers is not in the document]
Comment (maurelian) [reply .semgrepignore:23]: > If we want a newline at EOF

For the most part we've kept to it so far, and I've definitely had it nit'd on in the past.

> it should be a lint.

Do we currently have a linter that would work on every type of file we want to enforce the rule on?

File: .circleci/config.yml [hunk 2]
```diff
@@ -498,9 +498,39 @@ jobs:
           name: run itests
           command: make test-integration
 
+  semgrep-scan:
+    parameters:
+        diff_branch:
+          type: string
+          default: develop
+    environment:
+      # Scan changed files in PRs, block on new issues only (existing issues ignored)
```
[the remark this answers is not in the document]
Comment (maurelian) [reply .circleci/config.yml:507]: Oh, I see, the findings in [the report](https://app.circleci.com/pipelines/github/ethereum-optimism/optimism/2788/workflows/e2a028ec-a9ac-41af-a04e-65d2e0a19a6d/jobs/29758) are "non-blocking".

Is it also in an app on their site that the blocking status of each rule is defined?

[the remark this answers is not in the document]
Comment (trianglesphere) [reply .circleci/config.yml:507]: yea, most of the config is done in the app. I've got n in-progress notion writeup for this & that's gonna be the best place for information on semgrep.

**Pull request closed without merging**